store/tikv/split_region.go

```diff
@@ -157,28 +199,41 @@ func (s *KVStore) batchSendSingleRegion(bo *Backoffer, batch batch, scatter bool
 	if len(spResp.Regions) > 0 {
 		newRegionLeft = logutil.Hex(spResp.Regions[0]).String()
 	}
-	logutil.BgLogger().Info("batch split regions complete",
+	logutil.BgLogger().Info("batch split regions, split regions request complete",
 		zap.Uint64("batch region ID", batch.regionID.id),
+		zap.Int64("table ID", tid),
+		zap.Duration("cost-time", time.Since(start)),
 		zap.Stringer("first at", kv.Key(batch.keys[0])),
 		zap.String("first new region left", newRegionLeft),
 		zap.Int("new region count", len(spResp.Regions)))
 
-	if !scatter {
-		return batchResp
-	}
+	return batchResp
+}
 
-	for i, r := range spResp.Regions {
+func (s *KVStore) scatterRegionsInBatchSplitResp(bo *Backoffer, tableID *int64, batchResp singleBatchResp) singleBatchResp {
+	var err error
+	start := time.Now()
+	tid := int64(0)
+	if tableID != nil {
+		tid = *tableID
+	}
+	spResp := batchResp.resp.Resp.(*kvrpcpb.SplitRegionResponse)
+	for _, r := range spResp.Regions {
 		if err = s.scatterRegion(bo, r.Id, tableID); err == nil {
```

Comment:
So it's still using https://github.com/pingcap/tidb/blob/579421f6e02ab5cdde3344aa25815725a1561f12/store/tikv/split_region.go#L217 the batch API to scatter regions one by one. Which may make the scatter not work?

Reply:
I mean https://github.com/pingcap/tidb/issues/22788 .
OK, I'm off topic.

Reply:
The current region scatter may not work in a specific case which described in https://github.com/tikv/pd/pull/3422 ,  I think it's not related to the api.